pkg/deploy/api/types.go

```diff
@@ -182,6 +182,14 @@ type DeploymentTriggerImageChangeParams struct {
 	From kapi.ObjectReference `json:"from"`
 	// Tag is the name of an image repository tag to watch for changes.
 	Tag string `json:"tag,omitempty"`
+	// Status contains information about the last trigger event.
+	Status DeploymentTriggerImageChangeParamsStatus `json:"status"`
```

Comment:
Isn't BuildConfigTriggers the odd one out here? It has fields that are spec-like configuration supplied by user, and then also a status-like LastVersion set by the system which isn't meant to be set by users. I felt like I was correcting that here by encapsulating the read-only system-set fields into status.